executor/joiner.go

```diff
@@ -261,7 +261,7 @@ func (j *baseJoiner) filter(
 			innerColOffset, outerColOffset = len(lUsed), 0
 		}
 	}
-	return chunk.CopySelectedJoinRowsWithSameOuterRows(input, innerColOffset, outerColOffset, j.selected, output)
```

Comment:
I think we can keep this function here because it's effective.
We can add a check logic to determine whether we need to append the inner cols or outer cols.

Reply:
`chunk.CopySelectedJoinRowsWithSameOuterRows()`  just is more efficient in microbenchmark, but it makes little improvement when run real SQLs, especially on large datasets.

Here, I wonder how to keep the build/probe with used columns in accordance? This accordance decides whether the outer part of joined rows are the same or not.